alert-jira/src/main/java/com/synopsys/integration/alert/jira/common/JiraMessageContentConverter.java

```diff
@@ -68,44 +70,46 @@ public JiraMessageContentConverter(JiraMessageParser jiraMessageParser) {
             return List.of(issueRequest);
         } else {
             logger.debug("The top level action was not a DELETE action so it will be ignored");
         }
         return List.of();
     }
 
-    protected List<IssueTrackerRequest> createOrUpdateIssuesByComponentGroup(String providerName, String providerUrl, LinkableItem topic, LinkableItem nullableSubTopic,
-        SetMap<String, ComponentItem> groupedComponentItems)
-        throws IntegrationException {
+    protected List<IssueTrackerRequest> createOrUpdateIssuesByComponentGroup(ContentKey providerContentKey, String providerUrl, LinkableItem topic, LinkableItem nullableSubTopic,
+        SetMap<String, ComponentItem> groupedComponentItems) throws IntegrationException {
         List<IssueTrackerRequest> issues = new LinkedList<>();
 
         for (Set<ComponentItem> componentItems : groupedComponentItems.values()) {
             ComponentItem arbitraryItem = componentItems
                                               .stream()
                                               .findAny()
-                                              .orElseThrow(() -> new AlertException(String.format("No actionable component items were found. Provider: %s, Topic: %s, SubTopic: %s", providerName, topic, nullableSubTopic)));
+                                              .orElseThrow(
+                                                  () -> new AlertException(String.format("No actionable component items were found. Provider: %s, Topic: %s, SubTopic: %s", providerContentKey.getProviderName(), topic, nullableSubTopic)));
             ItemOperation operation = arbitraryItem.getOperation();
             String trackingKey = createAdditionalTrackingKey(arbitraryItem);
-            IssueSearchProperties issueSearchProperties = JiraIssuePropertiesUtil.create(providerName, providerUrl, topic, nullableSubTopic, arbitraryItem, trackingKey);
+            IssueSearchProperties issueSearchProperties = JiraIssuePropertiesUtil.create(providerContentKey.getProviderName(), providerUrl, topic, nullableSubTopic, arbitraryItem, trackingKey);
 
             IssueTrackerRequest issueRequest = null;
+            AlertIssueOrigin alertIssueOrigin = new AlertIssueOrigin(providerContentKey, arbitraryItem);
             if (ItemOperation.ADD == operation || ItemOperation.UPDATE == operation) {
-                IssueContentModel issueContentModel = jiraMessageParser.createIssueContentModel(providerName, IssueCreationRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
-                issueRequest = IssueCreationRequest.of(issueSearchProperties, issueContentModel);
+                IssueContentModel issueContentModel = jiraMessageParser.createIssueContentModel(providerContentKey.getProviderName(), IssueCreationRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
+                issueRequest = IssueCreationRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
```

Comment:
Would we be able to convert this to a private method that's used in these if checks?